### DISCUSSION TITLE
Weird Collection Name Due to Mongoose utils.toCollectionName

### DISCUSSION THREAD

**Discussion by rawkode** [payloadcms/payload#3103] [Feature Requests & Ideas]
### Link to reproduction

https://github.com/rawkode/payload/tree/bug/pluralize

### To Reproduce

Create a `CollectionConfig` with a slug of `people`.

### Describe the Bug

Inspect the mongoDB tables and one will be called "peoples"

I tracked this down to `utils.toCollectionName` in mongoose, which in turn calls `pluralize`, available [here](https://github.com/Automattic/mongoose/blob/master/lib/helpers/pluralize.js).

This could be by-passed by providing a variable on `CollectionConfig` to specify the collection name, which would then be passed to `Schema()` via [schemaOptions](https://github.com/Automattic/mongoose/blob/d8d48458507033b24f2a66fcb43577e2d0b95c61/types/schemaoptions.d.ts#L65)

### Payload Version

1.9.5

## Update from Payload - 31 July 2023

We have determined to add this as a feature request and solve it via a top-level collection property called `databaseEntityName` or similar, which will be enforced on a collection-by-collection basis.

**Comment by AlessioGr** [comment]
Related PR (not merged yet): https://github.com/payloadcms/payload/pull/2836

Might also disable this by default in 2.0, though that's not decided yet

**Comment by rawkode** [comment]
Maybe 2.0 would be a good opportunity to make the current config of:

```ts
export const People: CollectionConfig = {
	slug: "people",
	labels: { singular: "Person", plural: "People" },
	graphQL: { singularName: "Person", pluralName: "People" },
}
```

more streamlined:

```ts
export const People: CollectionConfig = {
	slug: "people",
        singular: "person",
        plural: "people",
	
        // Infer these from above
        // labels: { singular: "Person", plural: "People" },
	// graphQL: { singularName: "Person", pluralName: "People" },
}
```

**Comment by jmikrut** [comment]
OK so the team and I just talked about this a little bit - and as of now we are hoping to be as declarative as possible here, so that developers have as much flexibility as possible regarding `slug`, the database name, labels used in admin / error messages, etc.

We are also moving toward a database adapter pattern, meaning that any Mongoose-specific properties should be scoped to the future Mongoose adapter itself, and not present in the main Payload config. So that will have some ramifications regarding how we solve this.

We will definitely resolve this one way or another very soon. Thank you for bringing this up!

**Comment by lablancas** [comment]
@jmikrut I am working on a prototype to migrate to Payload with an existing MongoDB. I have been using camel case for my collection names, but it seems like the `slug` set for a CollectionConfig is being converted to lowercase due to the Mongoose model name conversion. Is there any changes coming that will allow for setting the collection name in the CollectionConfig to allow for overriding the conversion. Just checking before I look into renaming my collections.
